core/ir/instr_shared.c

```diff
@@ -2815,6 +2815,12 @@ decode_memory_reference_size(void *drcontext, app_pc pc, uint *size_in_bytes)
 uint
 instr_num_memory_read_access(instr_t *instr)
 {
+    /* DR_ISA_REGDEPS instructions don't have an opcode, hence we use their category to
+     * determine whether they perform at least one read.
+     */
+    if (instr->isa_mode == DR_ISA_REGDEPS)
+        return TESTANY(DR_INSTR_CATEGORY_LOAD, instr_get_category(instr)) ? 1 : 0;
```

Comment:
As of now `instr_num_memory_read_access()` and `instr_num_memory_write_access()` are only used in invariant_checker, however I wonder if this check should be only in invariant_checker, rather than here in core/ir/.

Reply:
It may be misleading to give an answer here which may be interpreted as precise.  I would move this to inside invariant checker.  Ditto for writes.

Reply:
Agreed, moved into invariant_checker. Done.